addons/website_sale/controllers/delivery.py

```diff
@@ -128,7 +135,13 @@ def website_sale_set_pickup_location(self, pickup_location_data):
         order_sudo = request.website.sale_get_order()
         order_sudo._set_pickup_location(pickup_location_data)
 
-    @route('/website_sale/get_pickup_locations', type='jsonrpc', auth='public', website=True)
+    @route(
+        '/website_sale/get_pickup_locations',
+        type='jsonrpc',
+        auth='public',
+        website=True,
+        readonly=True,
```

Comment:
NB: currently not readonly when click & collect is installed & running.
Will be fixed by #187608